tests/test_causal_graph.py

```diff
@@ -538,6 +538,16 @@ def test_adjacency_consistency(self):
 
         self.assertEqual(CausalGraph.from_adjacency_matrix(*cg.to_numpy()), cg)
 
+    def test_copy(self):
+        # test copy preserves the correct class
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_edge('a', 'b')
+        ts_cg.add_edge('a lag(n=1)', 'b')
+        ts_cg.add_edge('b lag(n=1)', 'b')
```

Comment:
Sorry one last thing. Would add some general meta on edges and nodes. Then test what happens when you say copy without meta. Is the TS graph still created correctly as we still need the variable name and time lag info but the other meta can be lost.